### PR TITLE
Deprecate Python 2 formatting support

### PR DESCRIPTION
**Pull request by ichard26**
### Description

It's 2021. See also #2251.

### Checklist - did you ...

- [x] Add a CHANGELOG entry if necessary?
- [x] Add / update tests if necessary?
- [x] Add new / update outdated documentation?

### CODE REVIEW

**Comment by ichard26** [comment]
Just tagging this as a "do not merge" as all of this is very much up for debate.

**Comment by ichard26** [comment]
The current structure of the commits require that this is merged via a rebase so the split ends up being useful :)

> **Prepare for Python 2 depreciation** 
>
> - Use BlackRunner and .stdout in command line test
>
> So the next commit won't break this test. This is in its own commit so we can just revert the depreciation commit when dropping Python 2 support completely.

**Review by cooperlees — APPROVED** [review]
LGTM - Lets merge?

**Review by graingert — CHANGES_REQUESTED** [review]
The comments aren't consistent

**Review by graingert — APPROVED** [review]

**Comment by ichard26** [comment]
One potential problem I do want to bring up is that right now each call to format_str with Python 2 warns ... that may be too noisy.

File: docs/faq.md [hunk 1]
```diff
@@ -74,13 +74,17 @@ disabled-by-default counterpart W504. E203 should be disabled while changes are
 
 ## Does Black support Python 2?
 
+```{warning}
+Python 2 support has been deprecated since 21.10b0.
+
+this support will be dropped in the first stable release, expected for January 2022.
```
Comment (JelleZijlstra) [inline docs/faq.md:80]:
```suggestion
This support will be dropped in the first stable release, expected for January 2022.
```

**Review by JelleZijlstra — COMMENTED** [review]
Noticed one typo.

I'm also still thinking about the feature detection stuff here and brought it up on Discord.

**Comment by JelleZijlstra** [comment]
Printing for each file might be noisy, but properly doing it only once might require communication across asyncio workers, which I don't really want to have to deal with.

**Review by JelleZijlstra — APPROVED** [review]
Let's merge it.

**Review by aarav2you — APPROVED** [review]

**Pull request merged**